src/test/scala/chiselTests/QueueFlushSpec.scala

```diff
@@ -0,0 +1,181 @@
+    assert(elems(outCnt.value) === q.io.deq.bits)
+    outCnt.inc()
+  }
+  when(outCnt.value === elements.length.U) {
+    stop()
+  }
+}
+
+class QueueGetsFlushedTester (elements: Seq[Int], queueDepth: Int, bitWidth: Int, tap: Int, useSyncReadMem: Boolean) extends BasicTester {
+    val q = Module(new Queue(UInt(bitWidth.W), queueDepth, hasFlush = true))
+    val elems = VecInit(elements.map(_.U))
+
+  val inCnt = Counter(elements.length + 1)
+  val outCnt = Counter(elements.length + 1)
+  
+  val halfCnt = (queueDepth + 1)/2
+  val cycleCounter = Counter(elements.length + 1)
+  //testing a flush when 
+  val flush = LFSR(16)((tap + 3) % 16)
+  q.io.flush.get := flush
+  q.io.enq.valid := (inCnt.value < elements.length.U)
+  q.io.deq.ready := LFSR(16)(tap)
+  cycleCounter.inc() //counts every cycle
+
+  q.io.enq.bits := elems(inCnt.value)
+  when(q.io.enq.fire()) {
+    inCnt.inc()
+  }
+  when(q.io.deq.fire()) {
+    outCnt.inc()
+  }
+
+  when(flush) {
+    //check that queue gets flushed
+    assert((q.io.count === 0.U) || q.io.deq.valid, s"in count: ${q.io.count} and queue length: $queueDepth" ) 
+    assert(!q.io.enq.ready, s"in count: ${q.io.count} and queue length: $queueDepth") 
+  } 
+  
+  when(inCnt.value === elements.length.U) { //stop when all entries are enqueued
+    stop()
+  }
+}
+
+class EmptyFlushEdgecaseTester (elements: Seq[Int], queueDepth: Int, bitWidth: Int, tap: Int, useSyncReadMem: Boolean) extends BasicTester {
+    val q = Module(new Queue(UInt(bitWidth.W), queueDepth, hasFlush = true))
+    val elems = VecInit(elements.map(_.U))
+
+  val inCnt = Counter(elements.length + 1)
+  val outCnt = Counter(elements.length + 1)
+  
+  val cycleCounter = Counter(elements.length + 1)
+  //testing a flush when queue is empty
+  val flush = (cycleCounter.value === 0.U && inCnt.value === 0.U) //flushed only before anything is enqueued  
+  q.io.flush.get := flush
+  cycleCounter.inc() //counts every cycle
+
+  q.io.enq.valid := (inCnt.value < elements.length.U)
+  q.io.deq.ready := LFSR(16)(tap)
+
+  q.io.enq.bits := elems(inCnt.value)
+  when(q.io.enq.fire()) {
+    inCnt.inc()
+  }
+  when(q.io.deq.fire()) {
+    outCnt.inc()
+  }
+
+  when(flush) {
+    //check that queue gets flushed at the beginning with no elements
+    assert(!q.io.deq.valid, s"in count: ${q.io.count} and queue length: $queueDepth" ) 
+    assert(!q.io.enq.ready, s"in count: ${q.io.count} and queue length: $queueDepth") 
+  } 
+  
+  when(inCnt.value === elements.length.U) { //stop when all entries are enqueued
+    stop()
+  }
+}
+
+class FullQueueFlushEdgecaseTester (elements: Seq[Int], queueDepth: Int, bitWidth: Int, tap: Int, useSyncReadMem: Boolean) extends BasicTester {
+    val q = Module(new Queue(UInt(bitWidth.W), queueDepth, hasFlush = true))
+    val elems = VecInit(elements.map(_.U))
+
+  val inCnt = Counter(elements.length + 1)
+  val currDepthCnt = Counter(queueDepth + 1)
+  
+  //testing a flush when queue is empty
```

Comment:
when queue is full in this case?

Reply:
i think this comment is still wrong
```suggestion
  //testing a flush when queue is full
```